### PR TITLE
Feat/359 sites aggregation

### LINKED ISSUE #359: [meta] Sites view round two: adding regional and dno features
This meta issue contains tasks for completing the Regional and DNO features of the `Sites View UI`. 

## Detailed Description
This is an evolving task list: 

- [ ] #360
- [ ] #361

## Context
When working on the `Sites View` in March, the Region and and DNO UI features were not fully completed.

### LINKED ISSUE #360: Sites View map slider

### LINKED ISSUE #361: Sites View regional and dno tables

### PR DESCRIPTION
**Pull request by braddf**
# Pull Request

## Description

Main objective: add DNO- and GSP-level aggregation to Sites view, automatically changing aggregation level in response to user zoom on map.
Also includes loading state additional functionality/type safety, and some general UI fixes/improvements.

Fixes #359 
Fixes #360 
Fixes #361 

## How Has This Been Tested?

Manual testing on branch deploy: https://nowcasting-app-git-feat-359-sites-aggregation-openclimatefix.vercel.app/

- [x] Yes

_If your changes affect data processing, have you plotted any changes? i.e. have you done a quick sanity check?_

- [x] Yes, aggregation headers and totals seem to make sense

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: apps/nowcasting-app/components/charts/solar-site-view/solar-site-chart.tsx [hunk 1]
```diff
@@ -141,12 +217,32 @@ const SolarSiteChart: FC<{
...
+      <div className="flex-auto flex flex-col mb-2">
         <div className="flex content-between bg-ocf-gray-800 h-auto">
-          <div className="text-white lg:text-2xl md:text-lg text-base font-black m-auto ml-5 flex justify-evenly">
+          <div className="text-white lg:text-2xl md:text-lg text-base font-black m-auto mx-3 flex justify-evenly">
             All Sites
           </div>
-          <div className="flex justify-between flex-2 my-2 px-6">
+          <div className="flex-1 flex flex-col items-center justify-around text-2xs py-2 leading-none">
+            {Object.entries(
+              sitesLoadingState?.endpointStates || ({} as LoadingState<SitesEndpointStates>)
```
Comment (rachel-labri-tipton) [inline apps/nowcasting-app/components/charts/solar-site-view/solar-site-chart.tsx:227]: Loading panel could be useful as a hover state. Could think about how this works with the "Loading" message.

File: apps/nowcasting-app/components/hooks/useFormatSitesData.tsx [hunk 2]
```diff
@@ -43,19 +46,18 @@ export const useFormatSitesData = (
...
-  // });
-
-  const data = useMemo(() => {
+  const firstNonZeroForecastValue =
+    combinedSitesData.sitesPvForecastData?.[0]?.forecast_values.find(
+      (fcv) => fcv.expected_generation_kw > 0.01
+    );
+  const firstForecastData = JSON.stringify(firstNonZeroForecastValue, null, 2) || "";
+  return useMemo(() => {
+    // const [sitesTableData, setSitesTableData] = useState<AggregatedSitesCombinedData>({
```
Comment (rachel-labri-tipton) [inline apps/nowcasting-app/components/hooks/useFormatSitesData.tsx:55]: Can this commented out code be removed?

File: apps/nowcasting-app/components/map/deltaMap.tsx [hunk 3]
```diff
@@ -188,46 +172,62 @@ const DeltaMap: React.FC<DeltaMapProps> = ({
...
-
-    map.current.addLayer({
-      id: "latestPV-forecast",
-      type: "fill",
-      source: "latestPV",
-      layout: { visibility: "visible" },
-      paint: {
-        "fill-color": getFillColor("delta"),
-        "fill-opacity": 0.7
+  const addOrUpdateFCData = (map: mapboxgl.Map) => {
```
Comment (rachel-labri-tipton) [inline apps/nowcasting-app/components/map/deltaMap.tsx:175]: each map has one of these now

File: apps/nowcasting-app/components/map/pvLatest.tsx [hunk 4]
```diff
@@ -115,67 +115,88 @@ const PvLatestMap: React.FC<PvLatestMapProps> = ({
     return generateGeoJsonForecastData(initForecastData, selectedISOTime);
   }, [initForecastData, selectedISOTime]);
 
-  const updateMapData = (map: mapboxgl.Map) => {
+  // const updateMapData = (map: mapboxgl.Map) => {
```
Comment (rachel-labri-tipton) [inline apps/nowcasting-app/components/map/pvLatest.tsx:118]: let's remove these comments too

File: apps/nowcasting-app/components/map/sitesMap.tsx [hunk 5]
```diff
@@ -509,190 +394,101 @@ const SitesMap: React.FC<SitesMapProps> = ({
...
+            theme.extend.colors["ocf-yellow"].DEFAULT || "#f9d71c"
+          ],
+          "circle-opacity": 0.8
+        }
+      });
+    }
+    console.log("end addOrUpdateMapGroup");
+  };
+
+  const addFCData = (map: mapboxgl.Map) => {
```
[the remark this answers is not in the document]
Comment (braddf) [reply apps/nowcasting-app/components/map/sitesMap.tsx:445]: Started doing this, but actually thought it makes sense to do this as a more involved refactor which might come out of the frontend testing and general tidy up I'm hoping we'll do next month, so going to leave as is for now 😊

File: apps/nowcasting-app/components/charts/solar-site-view/solar-site-chart.tsx [hunk 6]
```diff
@@ -181,34 +277,44 @@ const SolarSiteChart: FC<{
...
-              />
-            </div>
-          </>
+        {clickedSiteGroupId && aggregationLevel !== AGGREGATION_LEVELS.NATIONAL && (
+          <div className="flex-1 flex flex-col relative h-60 dash:h-auto">
+            <>
+              <ForecastHeaderSite
+                forecast={
+                  getTotalPvForecastGenerationForGroup(
+                    selectedSiteData.map((site) => site.site_uuid),
```
[the remark this answers is not in the document]
Comment (braddf) [reply apps/nowcasting-app/components/charts/solar-site-view/solar-site-chart.tsx:266]: Same as other comment, start but think best to refactor all of these charts/views at the same time in a uniform way, and potentially share some of these helper functions too where applicable

File: apps/nowcasting-app/components/map/sitesMap.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (rachel-labri-tipton) [reply apps/nowcasting-app/components/map/sitesMap.tsx:445]: good idea, @braddf. better to group all these refactors together.

**Review by rachel-labri-tipton — APPROVED** [review]
This looks good @braddf. I like the loading state for all the charts.

File: apps/nowcasting-app/components/map/map.tsx [hunk 7]
```diff
@@ -70,22 +72,46 @@ const Map: FC<IMap> = ({
...
+  // useEffect(() => {
+  //   if (!map.current) return; // wait for map to initialize
+  //
+  //   map.current.on("moveend", () => {
+  //     console.log("setting map state");
+  //     setLng(Number(map.current?.getCenter().lng.toFixed(4)));
+  //     setLat(Number(map.current?.getCenter().lat.toFixed(4)));
+  //     setZoom(Number(map.current?.getZoom().toFixed(2)));
+  //   });
+  // }, [map]);
```
Comment (peterdudfield) [inline apps/nowcasting-app/components/map/map.tsx:114]: remove comments

File: apps/nowcasting-app/pages/index.tsx [hunk 8]
```diff
@@ -298,45 +299,130 @@ export default function Home({ dashboardModeServer }: { dashboardModeServer: str
...
-    );
+  const {
+    data: sitePvForecastData,
+    isLoading: sitePvForecastLoading,
+    isValidating: sitePvForecastValidating,
+    error: sitePvForecastError
+  } = useLoadDataFromApi<SitesPvForecast>(
+    `${SITES_API_PREFIX}/sites/pv_forecast?site_uuids=${siteUuidsString}`,
+    {
+      // isPaused: () => {
```
Comment (peterdudfield) [inline apps/nowcasting-app/pages/index.tsx:330]: remove

File: apps/nowcasting-app/pages/index.tsx [hunk 9]
```diff
@@ -298,45 +299,130 @@ export default function Home({ dashboardModeServer }: { dashboardModeServer: str
...
+  const {
+    data: sitesPvActualData,
+    isLoading: sitePvActualLoading,
+    isValidating: sitePvActualValidating,
+    error: sitePvActualError
+  } = useLoadDataFromApi<SitesPvActual>(
     `${SITES_API_PREFIX}/sites/pv_actual?site_uuids=${siteUuidsString}`,
-    {}
+    {
+      // isPaused: () => {
```
Comment (peterdudfield) [inline apps/nowcasting-app/pages/index.tsx:349]: remove

**Pull request merged**